resources/splunk_service.rb

```diff
@@ -62,14 +65,30 @@ def install_state
   action_class do
     include CernerSplunk::ProviderHelpers
 
-    def service_action(desired_action)
+    # Provide file and service resources from methods to eliminate accidental resource cloning
+    def marker_resource
+      resources(file: marker_path.to_s)
+    rescue Chef::Exceptions::ResourceNotFound
+      file(marker_path.to_s) do
+        action :nothing
+      end
+    end
+
+    def service_resource
```

Comment:
I think you're suggesting declaring the service and file resources in every action they're used, which would equate to repeating those exact declarations with different actions 5 times (the linux specific implementation requires them to be redefined because the parent actions are not reused). In addition, I don't think it's actually doable. I tried declaring the resources normally in service_action (how it used to be, for the most part) but since I've started notifying actions on the resource's self, it is causing those to be cloned resources and throw warnings. The methods are defined in action_class so they should be scoped to the provider and declare those resources in the forked run context, but I don't know why each action run does not have its own run context. Regardless, the outcome is that all the actions being run on the service resource as part of checking and restarting off the restart marker are referencing the same run context so I wrote these methods to protect against resource cloning.